source/MapPlanetCard.cpp

```diff
@@ -0,0 +1,247 @@
+
+vector<MapPlanetCard *> MapPlanetCard::cards{};
+double MapPlanetCard::scroll = 0.;
+
+
+
+MapPlanetCard::MapPlanetCard(const StellarObject &object, bool hasVisited) : hasVisited(hasVisited), planetName(object.Name())
+{
+	number = cards.size();
+	cards.emplace_back(this);
+	const Planet *planet = object.GetPlanet();
+	hasSpaceport = planet->HasSpaceport();
+	hasShipyard = planet->HasShipyard();
+	hasOutfitter = planet->HasOutfitter();
+
+	reputationLabel = !hasSpaceport ? "No Spaceport" :
+		GameData::GetPolitics().HasDominated(planet) ? "Dominated" :
+		planet->GetGovernment()->IsEnemy() ? "Hostile" :
+		planet->CanLand() ? "Friendly" : "Restricted";
+	
+	sprite = object.GetSprite();
+
+	const Interface* planetCardInterface = GameData::Interfaces().Get("map planet card");
+	float planetIconMaxSize = static_cast<float>(planetCardInterface->GetValue("planet icon max size"));
+	spriteScale = min(.5f, min((planetIconMaxSize) / sprite->Width(), (planetIconMaxSize) / sprite->Height()));
+}
+
+
+
+MapPlanetCard::~MapPlanetCard()
+{
+	auto begin = cards.begin();
+	bool found = false;
+	unsigned i = 0;
+	for(auto &it : cards)
+	{
+		// The ones after the one we deleted need their number adjusted.
+		if(found)
+			--it->number;
+		else if(it == this)
+			found = true;
+		else
+			++i;
+	}
+	// We should be in it, but just to be sure.
+	if(found)
+		cards.erase(begin + i);
+}
+
+
+
+bool MapPlanetCard::Click(int x, int y, int clicks)
+{
+	// The isShown variable should have already updated by the drawing of this item.
+	if(isShown) {
+		const Interface *planetCardInterface = GameData::Interfaces().Get("map planet card");
+		double textStart = planetCardInterface->GetValue("text start");
+		double categorySize = planetCardInterface->GetValue("category size");
+		double categories = planetCardInterface->GetValue("categories");
+
+		// The yCoordinate refers to the center of this object.
+		double relativeY = y - yCoordinate;
+		if(relativeY > 0. && relativeY < AvailableSpace())
+		{
+			isSelected = true;
+
+			// The first category is the planet name and is not selectable.
+			if(relativeY > textStart + categorySize && relativeY < textStart + categorySize * categories)
+				selectedCategory = (relativeY - textStart - categorySize) / categorySize;
+		}
+		else
+			isSelected = false;
+	}
+	else
+		isSelected = false;
+	return isSelected;
+}
+
+
+
+bool MapPlanetCard::DrawIfFits(const Point &uiPoint)
+{
+	// Need to update this before checking if the element fits.
+	yCoordinate = uiPoint.Y();
+	isShown = Shown();
+	if(isShown)
+	{
+		const Font &font = FontSet::Get(14);
+		const Color &faint = *GameData::Colors().Get("faint");
+		const Color &dim = *GameData::Colors().Get("dim");
+		const Color &medium = *GameData::Colors().Get("medium");
+		const auto alignLeft = Layout(140, Truncate::BACK);
+
+		const Interface *planetCardInterface = GameData::Interfaces().Get("map planet card");
+		double height = planetCardInterface->GetValue("height");
```

Comment:
I use these values a lot, should they be made static and initiated by the mapdetailpanel?

Reply:
I made them static when possible in the functions themselves